build/github/cockroach-microbench-ci.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+# Directories and file names
+output_dir="./artifacts/microbench"
+cleaned_current_dir="$output_dir/current"
+cleaned_base_dir="$output_dir/base"
+
+# hardcoded the file name as of now since `compare` requires a particular format for the file name.
+benchmark_file_name="pkg_sql-report.log"
+log_output_file_path="$output_dir/microbench.log"
+storage_bucket_url="gs://cockroach-microbench-ci"
+
+# Threshold for comparison
+threshold=${COMPARE_THRESHOLD:-0}
+success_exit_status=0
+error_exit_status=1
+
+# Exit early if SKIP_COMPARISON is true and not a push step
+if $SKIP_COMPARISON && ! $PUSH_STEP; then
+  echo "Exiting since skip comparison is enabled and this is not a push step."
+  exit $success_exit_status
+fi
+
+# Build binary with Bazel
+bazel build --config crosslinux $(./build/github/engflow-args.sh) --jobs 100 //pkg/cmd/roachprod-microbench
+
+roachprod_microbench_dir="_bazel/bin/pkg/cmd/roachprod-microbench/roachprod-microbench_"
+
+mkdir -p "$output_dir"
+
+# Run benchmarks and clean output
+# running count=4 here because that's the minimum required for a comparison
+bazel test //pkg/sql/tests:all \
+  --config=use_ci_timeouts \
+  --jobs=300 \
+  --config=test \
+  --config=crosslinux \
+  --test_tmpdir=/tmp/cockroach \
+  --test_arg=-test.run=- \
+  --test_arg='-test.bench=^BenchmarkKV$' \
+  --test_sharding_strategy=disabled \
+  --test_arg=-test.cpu --test_arg=1 \
+  --test_arg=-test.v \
+  --test_arg=-test.count=4 \
+  --test_arg=-test.benchmem \
+  --crdb_test_off \
```

Comment:
Why are you doing `crdb_test_off` here? Is there some reason it's necessary? It's going to negatively affect caching.

Reply:
I ran a `dev bench` locally and it was present in that so used it here.

Reply:
Have removed it 👍🏼

Reply:
@rickystewart  I think `crdb_test_off` disables metamorphic tests (https://github.com/cockroachdb/cockroach/blob/master/pkg/util/metamorphic/constants.go#L97), which may taint the reproducibility of a microbenchmark if on, unless the seed is forced; which is maybe why it's part of the `dev bench` command and in other scripts that execute microbenchmarks.

Reply:
Check the same file for `DisableMetamorphicEnvVar`, i.e., to turn off metamorphic testing, set `COCKROACH_INTERNAL_DISABLE_METAMORPHIC_TESTING=true`. However, I think one reason that `crdb_test_off` is set in `dev bench` is that `crdb_test_off` also disables the testing assertions. I don't really know if this affects `sql_test` or not.